react/features/app/getRouteToRender.js

```diff
@@ -49,60 +49,71 @@ export function _getRouteToRender(stateful: Function | Object): Promise<Route> {
+    const joiningValidRoom = isRoomValid(room);
 
-            return;
+    // Mobile custom routing
+    if (isMobileApp) {
+        if (joiningValidRoom) {
+            route.component = Conference;
+        } else if (isWelcomePageAppEnabled(state)) {
+            route.component = WelcomePage;
         }
 
-        if (isRoomValid(room)) {
-            if (isMobileApp) {
-                route.component = Conference;
-                resolve(route);
-            } else {
-                // Update the location if it doesn't match. This happens when a
-                // room is joined from the welcome page. The reason for doing
-                // this instead of using the history API is that we want to load
-                // the config.js which takes the room into account.
-                const { locationURL } = state['features/base/connection'];
-
-                // eslint-disable-next-line no-negated-condition
-                if (window.location.href !== locationURL.href) {
-                    route.href = locationURL.href;
-                    resolve(route);
-                } else {
-                    // Maybe show deep-linking, otherwise go to Conference.
-                    getDeepLinkingPage(state).then(component => {
-                        route.component = component || Conference;
-                        resolve(route);
-                    });
-                }
-            }
+        return Promise.resolve(route);
+    }
 
-            return;
-        }
+    // Web custom routing when trying to join a meeting. There are essentially
+    // two possible handled routes here: welcome page and conference.
 
-        if (!isWelcomePageUserEnabled(state)) {
-            // Web: if the welcome page is disabled, go directly to a random
-            // room.
+    // We are intentionally not performing the check for mobile browsers because:
+    // - the WelcomePage is mobile ready;
+    // - if the URL points to a conference, getDeepLinkingPage will take
+    //   care of it.
+    const isUnsupportedBrowser
+        = !isMobileBrowser && !JitsiMeetJS.isWebRtcSupported();
 
-            let href = window.location.href;
+    if (joiningValidRoom) {
+        // Update the location if it doesn't match. This happens when a room is
+        // joined from the welcome page. The reason for doing this instead of
+        // using the history API is that we want to load the config.js which
+        // takes the room into account.
+        const { locationURL } = state['features/base/connection'];
 
-            href.endsWith('/') || (href += '/');
-            route.href = href + generateRoomWithoutSeparator();
-        } else if (isWelcomePageAppEnabled(state)) {
-            // Mobile: only go to the welcome page if enabled.
+        if (window.location.href !== locationURL.href) {
+            route.href = locationURL.href;
+
+            return Promise.resolve(route);
+        }
+
+        return getDeepLinkingPage(state)
```

Comment:
wouldn't this only apply to mobile?

Reply:
Deep linking applies to web, from my comprehension. There are two versions of deep linking here, one is on mobile browsers to link to the native app and the other is for any desktop browsers to link to a desktop app.